deltachat-ffi/src/lib.rs

```diff
@@ -2625,7 +2623,10 @@ pub unsafe extern "C" fn dc_msg_set_file(
         return;
     }
     let ffi_msg = &mut *msg;
-    ffi_msg.message.set_file(as_str(file), as_opt_str(filemime))
+    ffi_msg.message.set_file(
+        to_string_lossy(file),
+        to_opt_string_lossy(filemime).as_ref().map(|x| x.as_str()),
```

Comment:
This probably is safe, but I had to think about that.  I've still not studied or fully internalised the rules on when rust keeps objects alive though, only have a vague notion and know that it allows terrible mistakes with raw pointers in some cases.
I'd have stored the string in a local variable instead.

Reply:
maybe we should also provide a to_opt_path_lossy() function that does not crashes but returns None when utf-8 is bad in some way. but i wanted to fix this pr mainly the android crash and similar ffi functions.

there are also some as_str() calls eg. in the mime-parser that should probably be replaced the same way.